packages/core/saved-objects/core-saved-objects-server-internal/src/routes/index.ts

```diff
@@ -49,20 +50,30 @@ export function registerRoutes({
   migratorPromise: Promise<IKibanaMigrator>;
   kibanaVersion: string;
   kibanaIndex: string;
+  isServerless: boolean;
 }) {
   const router =
     http.createRouter<InternalSavedObjectsRequestHandlerContext>('/api/saved_objects/');
-  registerGetRoute(router, { config, coreUsageData, logger });
-  registerResolveRoute(router, { config, coreUsageData, logger });
-  registerCreateRoute(router, { config, coreUsageData, logger });
-  registerDeleteRoute(router, { config, coreUsageData, logger });
-  registerFindRoute(router, { config, coreUsageData, logger });
-  registerUpdateRoute(router, { config, coreUsageData, logger });
-  registerBulkGetRoute(router, { config, coreUsageData, logger });
-  registerBulkCreateRoute(router, { config, coreUsageData, logger });
-  registerBulkResolveRoute(router, { config, coreUsageData, logger });
-  registerBulkUpdateRoute(router, { config, coreUsageData, logger });
-  registerBulkDeleteRoute(router, { config, coreUsageData, logger });
+
+  const internalOnServerless = isServerless ? 'internal' : 'public';
```

Comment:
no. We cannot use `internalOnServerless` as is to globally define access for ALL the routes. If we do then we'd make every single one public on non-serverless!
We can't do that and still say we're removing them in v9.

Reply:
👍 , this is only used for the APIs documented as public, but that we want internal on serverless. I can add a comment to that effect.

Reply:
@jloleysens I'll approve this PR as long as we're not contradicting our public docs that clearly state which APIs are deprecated.